### ISSUE TITLE
Incompatibilities between RabbitMq v3 and v4

### ISSUE THREAD

**Issue by Knalltuete5000** [ansible-collections/community.rabbitmq#181]
##### SUMMARY

Since RabbitMq v4 is out there are incompatibilities between v3 and v4. This affects especially the `rabbitmq_upgrade` module since the `await_online_synchronized_mirror` command was removed in version 4.

##### ISSUE TYPE

Incompatibilities between version 3 and 4 which need to be fixed

##### COMPONENT NAME

Currently only the rabbitmq_upgrade module has been identitifed

##### COLLECTION VERSION

```
main -branch
```

##### OS / ENVIRONMENT

all os/environemts running rabbitmq v4

##### STEPS TO REPRODUCE

Run the module `rabbitmq_upgrade` with the action `await_online_synchronized_mirror` against a rabbitmq 4 server

##### EXPECTED RESULTS

Warning or failure because the command is not supported by the rabbitmq 4 server. Currently only the command output is presented to the user.

The module can or should check if the command is supported on the server and fail with an appropriate message

##### ACTUAL RESULTS

The command output from executing the command against a rabbitmq version 4 server

**Comment by Andersson007** [comment]
@Knalltuete5000 thanks a lot!
So just to put it straight, I'm not a rabbitmq user, never used it and i've no idea what's going on there technology specific. Just helping here folks:)

1. Do I understand correctly that our test playbooks install just the latest version of rabbitmq? If yes, the problem is that we can't start implementing RabbitMQ 4 support in modules **not testing** against partly not backwards compatible version 3. 
2. Testing against version 3 should be somehow added to the test setup. One solution to this can be adding another Ubuntu distro + variables files, and corresponding conditions `if it's ubuntu A, install RabbitMQ 4; elif it's ubuntu B, install RabbitMQ 3`. Then we can use similar conditions in the tests you changed, i.e. `run that commented task only against version 3`, etc.
3. The example can be found in https://github.com/ansible-collections/community.postgresql/tree/main/tests/integration/targets/setup_postgresql_db/vars
4. In theory, we could try to determine the version in the modules themselves and use simple conditions in their code but in this case we won't be sure that things still work against version 3.

In conclusion, adding testing for versions 3 and 4 prior to handling the incompatibilities would be highly recommended, though handling them with simple version based conditions in the code would be much better than nothing.

It's only my opinion. @Knalltuete5000 @csmart @shtlrs thoughts? Any feedback/help are much appreciated
